pkg/aws/ec2/api/cleanup/eni/eni_cleanup.go

```diff
@@ -100,79 +103,80 @@ func (e *ENICleaner) Start(ctx context.Context) error {
 // In the second cycle we can conclude that Interface 2 and 3 are leaked because they have been sitting for the time
 // interval between cycle 1 and 2 and hence can be safely deleted. And we can also conclude that Interface 1 was
 // created but not attached at the the time when 1st cycle ran and hence it should not be deleted.
-func (e *ENICleaner) cleanUpAvailableENIs() {
-	describeNetworkInterfaceIp := &ec2.DescribeNetworkInterfacesInput{
-		Filters: []*ec2.Filter{
-			{
-				Name:   aws.String("status"),
-				Values: []*string{aws.String(ec2.NetworkInterfaceStatusAvailable)},
-			},
-			{
-				Name:   aws.String("tag:" + e.clusterNameTagKey),
-				Values: []*string{aws.String(config.ClusterNameTagValue)},
-			},
-			{
-				Name: aws.String("tag:" + config.NetworkInterfaceOwnerTagKey),
-				Values: aws.StringSlice([]string{config.NetworkInterfaceOwnerTagValue,
-					config.NetworkInterfaceOwnerVPCCNITagValue}),
-			},
-		},
-	}
-
+func (e *ClusterENICleaner) DeleteLeakedResources() error {
+	var errors []error
+	var nextToken *string
+	eniFilters := e.GetENITagFilters()
 	availableENIs := make(map[string]struct{})
-
+	api.VpcCniAvailableENICnt.Set(0)
+	api.VpcRcAvailableENICnt.Set(0)
+	api.LeakedENIClusterCleanupCnt.Set(0)
 	for {
```

Comment:
Do we need a jitter in the loop to flatten possible API requests peak on EC2? Since this is in a cleanup process, a small delay between every request is ok and it could help protecting the account quota.

Reply:
Fixed by using the paginated API call with jitter